File: src/com/example/main/homework2/gambling/Rank.java
package com.example.main.homework2.gambling;

public enum Rank {
    TWO("2", 2),
    THREE("3", 3),
    FOUR("4", 4),
    FIVE("5", 5),
    SIX("6", 6),
    SEVEN("7", 7),
    EIGHT("8", 8),
    NINE("9", 9),
    JACK("Jack", 10),
    QUEEN("Queen", 11),
    KING("King", 12),
    ACE("Ace", 13),
    JOKER("Joker", 14);

    private final String name;
    private final int value;

    Rank(String name, int value) {
        this.name = name;
        this.value = value;
    }

    public String getName() {
        return name;
    }

    public int getValue() {
        return value;
    }

    public static Rank fromValue(int value) {
        for (Rank rank : Rank.values()) {
            if (value == rank.value) {
                return rank;
            }
        }
        throw new IllegalArgumentException("No rank with value " + value + " found");
    }
}

File: src/com/example/main/homework3/gambling/Rank.kt
package com.example.main.homework3.gambling

import java.lang.IllegalArgumentException

enum class Rank(val value: Int) {
    TWO(2),
    THREE(3),
    FOUR(4),
    FIVE(5),
    SIX(6),
    SEVEN(7),
    EIGHT(8),
    NINE(9),
    JACK(10),
    QUEEN(11),
    KING(12),
    ACE(13),
    JOKER(14);

    companion object {
        fun fromValue(value: Int) : Rank {
            for (rank in values()) {
                if (value == rank.value) {
                    return rank
                }
            }
            throw IllegalArgumentException("No rank with value $value found")
        }
    }
}